railties/lib/rails/engine.rb

```diff
@@ -728,9 +728,9 @@ def _all_load_paths(add_autoload_paths_to_load_path)
         end
       end
 
-      def fixtures_in_root_and_not_in_vendor?(fixtures)
+      def fixtures_in_root_and_not_in_gem_path?(fixtures)
         fixtures.exist? && fixtures.to_s.start_with?(Rails.root.to_s) &&
-          !fixtures.to_s.start_with?(Rails.root.join("vendor").to_s)
+          !Gem.path.any? { |gem_path| fixtures.to_s.start_with?(gem_path) }
```

Comment:
This doesn't work for what we want. Our gems are outside of vendor but inside `Gem.path`. This change will break our setup. It seems like dotenv should be installing the gem in the `vendor` folder.

Reply:
Another option is to exclude directories that start with `.`

Reply:
Oops! What do you think about adding a configurable list of excluded directories? 
This would allow us to the hard coded reference.

I could try to add something lightweight later this week.

Reply:
Hey @rafaelfranca. I started a brand new project ("rails new"), and I believe the default directory for bundler is .bundle (at least that is where mine were installed). Wonder if it would be worth it for rails new to set the bundle config path to vendor if that is the expected behavior (and allowing anyone who doesn't want that to override it). Haven't dug into all the issues tied to this, but I did bump into it on a pretty fresh run off the main branch. Might keep some people from having some headaches. If it makes sense, happy to take a look and push an PR for the default path. Thanks for all ya do!

Reply:
@phChun Please investigate a PR. 👍